addons/mail/models/mail_thread.py

```diff
@@ -796,43 +787,40 @@ def _create_bounce_email():
 
         def _warn(message):
             _logger.info('Routing mail with Message-Id %s: route %s: %s',
-                         message_id, route, message)
-
-        # Wrong model
-        if model and model not in self.pool:
-            if assert_model:
-                assert model in self.pool, 'Routing: unknown target model %s' % model
-            _warn('unknown target model %s' % model)
-            return ()
+                         message.get('Message-Id'), route, message)
 
-        # Private message: should not contain any thread_id
-        if not model and thread_id:
-            if assert_model:
-                if thread_id:
+        # Private message: should not contain any thread_id, should have a parent_id (only answers)
+        if not model:
+            if not thread_id:
+                if assert_model:
                     raise ValueError('Routing: posting a message without model should be with a null res_id (private message), received %s.' % thread_id)
-            _warn('posting a message without model should be with a null res_id (private message), received %s resetting thread_id' % thread_id)
-            thread_id = 0
-        # Private message: should have a parent_id (only answers)
-        if not model and not message_dict.get('parent_id'):
-            if assert_model:
-                if not message_dict.get('parent_id'):
+                _warn('posting a message without model should be with a null res_id (private message), received %s resetting thread_id' % thread_id)
+                thread_id = 0
+            if not message_dict.get('parent_id'):
+                if assert_model:
                     raise ValueError('Routing: posting a message without model should be with a parent_id (private mesage).')
-            _warn('posting a message without model should be with a parent_id (private mesage), skipping')
-            return False
-
-        if model and thread_id:
-            record_set = self.env[model].browse(thread_id)
-        elif model:
+                _warn('posting a message without model should be with a parent_id (private mesage), skipping')
```

Comment:
Typo.

Also maybe fold `assert_model`'s behaviour into `_warn` to avoid e.g. duplicating messages? Same below.